File: Jutge/2 - Divide and Conquer/1 P81966 Dichotomic search/main.c
#include <stdio.h>

int position(double x, double v[], int esq, int dre);

int main() {
	int n, t;
	while (scanf("%d", &n) > 0) {
    double v[n];
    for (int i = 0; i < n; ++i) scanf("%lf", &v[i]);
    scanf("%d", &t);
    while (t) {
      --t;
      double x;
      int esq, dre;
      scanf("%lf %d %d", &x, &esq, &dre);
      printf("%d\n", position(x, v, esq, dre));
    }
	}
	return 0;
}

File: Jutge/2 - Divide and Conquer/1 P81966 Dichotomic search/main.py
from jutge import read

n = read(int)
while n is not None:
  v = []
  for i in range(0, n):
    v.append(read(float))
  t = read(int)
  while t != 0:
    t -= 1
    x, esq, dre = read(float, int, int)
    print(position(x, v, esq, dre))
  n = read(int)

